File: app/src/main/java/com/example/administrator/androidtest/demo/BizDemo/TikTokDemo/Detail/TikTokVideoDetailActivity.java
package com.example.administrator.androidtest.demo.BizDemo.TikTokDemo.Detail;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;

import androidx.annotation.Nullable;

import com.bear.libcomponent.component.ComponentActivity;
import com.example.administrator.androidtest.databinding.ActTiktokVideoDetailBinding;

public class TikTokVideoDetailActivity extends ComponentActivity<ActTiktokVideoDetailBinding> {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        regActComponent(new VideoPlayComponent(getLifecycle()));
        regActComponent(new AdapterComponent(getLifecycle()));
    }

    @Override
    protected ActTiktokVideoDetailBinding inflateViewBinding(LayoutInflater inflater) {
        return ActTiktokVideoDetailBinding.inflate(inflater);
    }

    public static void go(Context context) {
        Intent intent = new Intent(context, TikTokVideoDetailActivity.class);
        context.startActivity(intent);
    }
}

File: app/src/main/java/com/example/administrator/androidtest/demo/bizdemo/tiktokdemo/detail/TikTokVideoDetailActivity.kt
package com.example.administrator.androidtest.demo.bizdemo.tiktokdemo.detail

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import com.bear.libcomponent.host.ComponentActivity
import com.example.administrator.androidtest.databinding.ActTiktokVideoDetailBinding
import com.example.administrator.androidtest.demo.ext.setupNormalSystemBar

class TikTokVideoDetailActivity : ComponentActivity<ActTiktokVideoDetailBinding>() {
    companion object {
        @JvmStatic
        fun go(context: Context) {
            val intent = Intent(context, TikTokVideoDetailActivity::class.java)
            context.startActivity(intent)
        }
    }

    override fun inflateViewBinding(inflater: LayoutInflater) = ActTiktokVideoDetailBinding.inflate(inflater)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupNormalSystemBar(lightStatusBars = false, lightNavigationBars = false)
        regComponent(ToolbarComponent())
        regComponent(AdapterComponent())
        regComponent(DataLoaderComponent())
    }
}
